rego/example_test.go

```diff
@@ -538,6 +538,8 @@ func ExampleRego_Partial() {
 	// Define a simple policy for example purposes.
 	module := `package test
 
+	default allow = false
```

Comment:
The `default` keyword implies ordering. I.e., the default rule is only defined if none of the other rules with the same name are defined. In the general case, the partial evaluation has to generate support rules to handle this.

For the initial Compile API (and `rego.Rego#Partial`) use cases where we're returning the partial evaluation result to the client, I think it's better if we do not have to deal with support rules.

In the short term, I think we can introduce a special case into partial evaluation where expressions like `data.example.allow = true` do not require a support rule when `data.example.allow` refers to a rule with a default value.